### ISSUE TITLE
False no warning when propname contains _handle

### ISSUE THREAD

**Issue by Rejpal** [yannickcr/eslint-plugin-react#1063]
When you have component like:

```
<Component
  onClickSomethingInComponent={this.method}
/>
```
You will get correct warning:
Handler function for onClickSomethingInComponent prop key must begin with '_handle'.

but if you correct it like this:

```
<Component
  _handleClickSomethingInComponent={this.method}
/>
```

you won't get the warning any more... which is wrong because the wrong thing here is that this.method still does not contain '_handle'.

correct state is 

```
<Component
  onClickSomethingInComponent={this._handleMethod}
/>
```

not the previous one... am I right here?

**Comment by ljharb** [comment]
Why would a handler function need to begin with "_handle"? Can you share your eslint config?

**Comment by ljharb** [comment]
Also, what is `Component` here?
